### DISCUSSION TITLE
Variables in the composition of materials UPBGE 0.2.5

### DISCUSSION THREAD

**Discussion by HolandonHdunto** [UPBGE/upbge#1535] [Ideas/Features for UPBGE 0.2.5+]
I was seeing the possibility of having variables in the composition of materials as in other game engines, in upbge there is LAMP-DATA in the case of LAMP-DATA even duplicating or grouping objects, the lamps despite being duplicated objects, the materials do not select the new ones duplicate lamps, the material continues with the initial lamp.
![LampData](https://user-images.githubusercontent.com/89164699/129977066-ef82578e-8d97-4636-a7f6-900e61919dd5.gif)
https://youtu.be/LKNf96yicgc

**Comment by BluePrintRandom** [comment]
this is a limitation of the old forward render

the lamps can't be instanced and they use resources even on a hidden layer

**Comment by HolandonHdunto** [comment]
It's a shame because Unity has these material properties to make effects. The only way to control materials in upbge is using Color Object RGB or lamps that don't work in groups.
